serial.ts

```diff
@@ -295,20 +335,20 @@ export class SerialPort {
    * closed.
    */
   public async close(): Promise<void> {
-    if (!this.device_.opened) {
-      throw new DOMException('InvalidStateError', 'Port not open.');
-    }
-
     const promises = [];
-    if (this.readable) {
-      promises.push(this.readable.cancel());
+    if (this.readable_) {
+      promises.push(this.readable_.cancel());
     }
-    if (this.writable) {
-      promises.push(this.writable.abort());
+    if (this.writable_) {
+      promises.push(this.writable_.abort());
     }
     await Promise.all(promises);
-    await this.setSignals({dtr: false});
-    await this.device_.close();
+    this.readable_ = null;
+    this.writable_ = null;
+    if (this.device_.opened) {
+      await this.setSignals({dtr: false});
```

Comment:
I would think that is we are closing this port then we are also not ready to send either.  Should we also set RTS to false?

Reply:
Done.